### PR TITLE
Add support for custom fields (lookup fields)

### PR DESCRIPTION
**Pull request by Vi6hal**
# Description of change
- Changelog & Version bump
- PR #124 

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

Why `interger` type for lookup fields and not `string`
Ans: [here](https://developer.zendesk.com/documentation/ticketing/using-the-zendesk-api/retrieving-lookup-relationship-fields-with-the-api/#:~:text=the%20id%20of%20the%20lookup%20relationship%20field.%20You%20can%20get%20the%20id%20from%20the%20API%20or%20from%20the%20URL%20of%20the%20custom%20field%27s%20page%20in%20the%20Zendesk%20interface.%20In%20the%20following%20example%2C%20the%20field%20id%20is%206498722764059%3A)

Co-author - @lukas-gust

### CODE REVIEW

**Review by rdeshmukh15 — APPROVED** [review]

**Review by nitingaikwad1 — APPROVED** [review]

**Review by cosimon — CHANGES_REQUESTED** [review]
Can you add the new field to the all_fields test?

**Comment by Vi6hal** [comment]
> Can you add the new field to the all_fields test?

This is not exactly a new field, but rather a custom field type, and I have setup lookup fields on few streams to validate lookup datatype.

I was thinking to add a new test case for the following reason:
- This type of field is not available on the UI for selection (being a nested field) although the parent object is available for selection
- all_fields_test only covers the fields in the root hierarchy of the object and does not cover nested fields/schema
- the location of lookup fields varies for streams, hence its preferable to validate it separately. 

@cosimon what are your thoughts about adding a new test?

**Comment by lukas-gust** [comment]
Let me know if I can be of any assistance.

**Comment by Vi6hal** [comment]
> Let me know if I can be of any assistance.

I have made some changes as suggested, once the team reviews and approves this change, i will merge this P.R.

File: test/unittests/test_customfield_schema_type.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    """
+
+    def get_z_field_obj(self, *params):
+        field = types.SimpleNamespace()
+        field.title, field.key, field.type = params
+        return field
+
+    def test_return_field_type_lookup(self):
+        expected_singer_type = {"type" : ["integer", "null"]}
+        z_field = self.get_z_field_obj("test_lookup_type","user_lookup_1","lookup")
```
Comment (RushiT0122) [inline test/unittests/test_customfield_schema_type.py:19]:
```suggestion
        z_field = self.get_z_field_obj("test_lookup_type", "user_lookup_1", "lookup")
```

File: tap_zendesk/streams.py [hunk 2]
```diff
@@ -39,10 +40,9 @@ def process_custom_field(field):
     zendesk_type = field.type
     json_type = CUSTOM_TYPES.get(zendesk_type)
     if json_type is None:
-        raise Exception("Discovered unsupported type for custom field {} (key: {}): {}"
-                        .format(field.title,
-                                field.key,
-                                zendesk_type))
+        json_type = "string"
+        LOGGER.critical("Discovered unsupported type for custom field %s (key: %s): %s",
+                    field.title, field.key, zendesk_type)
```
Comment (RushiT0122) [inline tap_zendesk/streams.py:45]:
```suggestion
    if zendesk_type not in CUSTOM_TYPES:
        LOGGER.critical("Discovered unsupported type for custom field %s (key: %s): %s",
                        field.title, field.key, zendesk_type)
    json_type = CUSTOM_TYPES.get(zendesk_type) or "string"
```

File: test/test_all_streams.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def rate_tickets(self, records):
             #zenpy_client.tickets.rate(id, rating) # example rating {'score': 'good'}
 
 
-    @ZendeskTest.skipUntilDone("TDL-20862")
+
```
Comment (RushiT0122) [inline test/test_all_streams.py:114]: Remove extra line here and other similar locations.

**Review by cosimon — APPROVED** [review]

**Review by RushiT0122 — APPROVED** [review]

**Pull request merged**